src/api/buildSwapTransaction.ts

```diff
@@ -30,19 +30,19 @@ export async function buildSwapTransaction(
   if ((apiParamsOrError as APIError).error) {
     return apiParamsOrError as APIError;
   }
-  let apiParams = apiParamsOrError as SwapAPIParams;
+  const apiParams = apiParamsOrError as SwapAPIParams;
 
   if (!params.useAggregator) {
-    apiParams = {
-      v2Enabled: true,
-      ...apiParams,
-    };
+    apiParams.v2Enabled = true;
```

Comment:
yeah sorry, my bad, let's go back to the 
```
{
      v2Enabled: true,
      ...apiParams,
    };
```
approach

Reply:
done

Reply:
```
  let apiParams = apiParamsOrError as SwapAPIParams;
  if (!params.useAggregator) {
    apiParams = {
      v2Enabled: true,
      ...apiParams,
    };
  }
  if (params.maxSlippage) {
    let slippagePercentage = params.maxSlippage;
    // Adjust slippage for V1 API (aggregator)
    // V1 expects slippage in tenths of a percent (e.g., 30 = 3%)
    if (params.useAggregator) {
      slippagePercentage = (Number(params.maxSlippage) * 10).toString();
    }
    apiParams = {
      slippagePercentage,
      ...apiParams,
    };
  }
```